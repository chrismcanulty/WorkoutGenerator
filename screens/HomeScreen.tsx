import {NativeStackHeaderProps} from '@react-navigation/native-stack';
import {View, Text} from 'react-native';
import React, {useEffect} from 'react';
import SplashScreen from 'react-native-splash-screen';

export default function HomeScreen({navigation}: NativeStackHeaderProps) {
  useEffect(() => {
    SplashScreen.hide();
  });
  return (
    <View>
      <Text>I am home screen</Text>
    </View>
  );
}
