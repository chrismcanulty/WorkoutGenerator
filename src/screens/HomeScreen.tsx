import {NativeStackHeaderProps} from '@react-navigation/native-stack';
import {ActivityIndicator, StyleSheet, View} from 'react-native';
import React, {useContext, useEffect, useState} from 'react';
// import AsyncStorage from '@react-native-async-storage/async-storage';
import styled from 'styled-components/native';
import questions from '../utils/questions';
import warnings from '../utils/warnings';
import {UserContext} from '../context/User.Context';
import {MultipleSelectList} from '../component/MultipleSelectList';
import {useFocusEffect} from '@react-navigation/native';

const Button = styled.TouchableOpacity`
  font-size: 24px;
  padding: 10px;
`;
const ButtonText = styled.Text`
  border: 2px rgb(230, 230, 230);
  border-radius: 15px;
  color: rgb(38, 38, 38);
  font-family: 'Montserrat-Regular';
  font-size: 24px;
  margin-left: 10px;
  margin-right: 10px;
  padding: 10px;
  text-align: center;
`;
const Header = styled.Text`
  color: rgb(38, 38, 38);
  font-family: 'Montserrat-Bold';
  font-size: 24px;
  margin: 20px;
  margin-top: 30px;
  padding: 18px;
  text-align: center;
`;
const Warning = styled.Text`
  border: 2px solid red;
  border-radius: 10px;
  color: red;
  font-family: 'Montserrat-Regular';
  font-size: 12px;
  margin: 20px;
  margin-top: 0px;
  padding: 10px;
  text-align: center;
`;
const ButtonWrapper = styled.View`
  background-color: white;
  bottom: 0;
  position: absolute;
  width: 100%;
`;
const ContainerWrapper = styled.ScrollView`
  background-color: white;
  height: 100%;
`;
const SpinnerWrapper = styled.View`
  background-color: white;
  flex: 1;
  justify-content: center;
  align-item: center;
`;

export default function HomeScreen({navigation}: NativeStackHeaderProps) {
  const {muscleGroup, selectedMuscles, setSelectedMuscles, loading} =
    useContext(UserContext);
  const [warning, setWarning] = useState(false);
  // Clear onboarding function and button - keep for testing purposes to test onboarding flow
  // const clearOnboarding = async () => {
  //   try {
  //     await AsyncStorage.removeItem('@isOnboarding');
  //   } catch (err) {
  //     console.log('Error @clearOnboarding: ', err);
  //   }
  // };

  const muscleGroupData = () => {
    const temp = [];
    for (const muscle of muscleGroup as any) {
      if (muscle.name_en) {
        temp.push({...muscle, key: muscle.id, value: muscle.name_en});
      }
    }
    return temp;
  };

  const onPress = function () {
    if (selectedMuscles.length > 0) {
      navigation.push('Equipment');
    } else if (selectedMuscles.length === 0) {
      setWarning(true);
    }
  };

  if (loading) {
    return (
      <SpinnerWrapper>
        <ActivityIndicator size="small" color="#0000ff" />
      </SpinnerWrapper>
    );
  }

  return (
    <>
      <ContainerWrapper>
        <Header>{questions[0]}</Header>
        <View>
          <MultipleSelectList
            setSelected={setSelectedMuscles}
            data={muscleGroupData()}
            save="value"
            boxStyles={styles.boxStyles}
            dropdownStyles={styles.dropdownStyles}
            dropdownTextStyles={styles.dropdownTextStyles}
            onSelect={() => setWarning(false)}
            label="Muscle groups"
            defaultValues={selectedMuscles}
          />
          {warning && <Warning>{warnings[0]}</Warning>}
        </View>
      </ContainerWrapper>
      <ButtonWrapper>
        <Button onPress={onPress}>
          <ButtonText>Next</ButtonText>
        </Button>
        {/* <Button onPress={clearOnboarding}>
          <ButtonText>Clear onboarding</ButtonText>
        </Button> */}
      </ButtonWrapper>
    </>
  );
}

const styles = StyleSheet.create({
  boxStyles: {
    marginLeft: 20,
    marginRight: 20,
    borderRadius: 10,
    borderWidth: 2,
    borderColor: 'rgb(230, 230, 230)',
  },
  dropdownStyles: {
    marginLeft: 20,
    marginRight: 20,
    marginBottom: 10,
    borderRadius: 10,
    borderWidth: 2,
    borderColor: 'rgb(230, 230, 230)',
  },
  dropdownTextStyles: {
    color: 'rgb(38, 38, 38)',
    fontFamily: 'Montserrat-Regular',
  },
});
