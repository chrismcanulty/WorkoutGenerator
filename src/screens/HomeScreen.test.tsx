import React from 'react';
import HomeScreen from './HomeScreen';
import renderer, {act} from 'react-test-renderer';
import {UserContext} from '../context/User.Context';
import {expect, test} from '@jest/globals';

const createTestProps = (props: Object) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

let props: any;
beforeEach(() => {
  props = createTestProps({});
});

test('renders correctly', async () => {
  const tree = renderer
    .create(
      <UserContext.Provider
        value={{muscleGroup: [], selectedMuscles: [], loading: false}}>
        <HomeScreen {...props} />
      </UserContext.Provider>,
    )
    .toJSON();
  await wait();
  await wait();
  expect(tree).toMatchSnapshot();
});

export async function wait(ms = 0) {
  await act(async () => {
    return new Promise(resolve => {
      setTimeout(resolve, ms);
    });
  });
}
