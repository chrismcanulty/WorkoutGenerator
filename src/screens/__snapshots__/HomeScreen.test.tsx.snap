// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <RCTScrollView
    style={
      [
        {
          "backgroundColor": "white",
          "height": "100%",
        },
      ]
    }
  >
    <View>
      <Text
        style={
          [
            {
              "color": "rgb(38, 38, 38)",
              "fontFamily": "Montserrat-Bold",
              "fontSize": 24,
              "marginBottom": 20,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 30,
              "paddingBottom": 18,
              "paddingLeft": 18,
              "paddingRight": 18,
              "paddingTop": 18,
              "textAlign": "center",
            },
          ]
        }
      >
        What muscle group do you want to target?
      </Text>
      <View>
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "borderColor": "rgb(230, 230, 230)",
                "borderRadius": 10,
                "borderWidth": 2,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 10,
                "marginLeft": 20,
                "marginRight": 20,
                "opacity": 1,
                "paddingHorizontal": 20,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "fontFamily": undefined,
                  },
                  undefined,
                ]
              }
            >
              Select option
            </Text>
            <Image
              resizeMode="contain"
              source={
                {
                  "testUri": "../../../assets/images/chevron.png",
                }
              }
              style={
                {
                  "height": 20,
                  "width": 20,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
  <View
    style={
      [
        {
          "backgroundColor": "white",
          "bottom": 0,
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "fontSize": 24,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          [
            {
              "borderColor": "rgb(230, 230, 230)",
              "borderRadius": 15,
              "borderStyle": "solid",
              "borderWidth": 2,
              "color": "rgb(38, 38, 38)",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 24,
              "marginLeft": 10,
              "marginRight": 10,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
              "textAlign": "center",
            },
          ]
        }
      >
        Next
      </Text>
    </View>
  </View>,
]
`;
